File: Objective-C Version/AnimatedCircleDemo/AnimatedCircleDemo/CircleView/CircleView.m
//
//  CircleView.m
//  AnimatedCircleDemo
//
//  Created by Kitten Yang on 7/23/15.
//  Copyright (c) 2015 Kitten Yang. All rights reserved.
//

#import "CircleView.h"

@implementation CircleView

+ (Class)layerClass{
    return [CircleLayer class];
}

-(id)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        self.circleLayer = [CircleLayer layer];
        self.circleLayer.frame = CGRectMake(0, 0, frame.size.width, frame.size.height);
        self.circleLayer.contentsScale = [UIScreen mainScreen].scale;
        [self.layer addSublayer:self.circleLayer];
    }
    return self;
}

@end

File: Swift Version/AnimatedCircleDemo-Swift/AnimatedCircleDemo-Swift/Classes/CircleView.swift
//
//  CircleView.swift
//  AnimatedCircleDemo-Swift
//
//  Created by Kitten Yang on 1/18/16.
//  Copyright © 2016 Kitten Yang. All rights reserved.
//

import UIKit

class CircleView: UIView {

    var circleLayer = CircleLayer()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        circleLayer.frame = CGRectMake(0, 0, frame.size.width, frame.size.height)
        circleLayer.contentsScale = UIScreen.mainScreen().scale
        layer.addSublayer(circleLayer)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
